### PR TITLE
feat:  correct monarchs.json and add source information

### LINKED ISSUE #595: Correct errors in monarchs.json or document them in SOURCES.md?
There are a couple of factual errors in the (as-of-yet unsourced) timeline in `monarchs.json` posted by @arvind in 2015. Since [README.md](https://github.com/vega/vega-datasets/blob/main/README.md) notes that

>*Datasets may contain intentional inconsistencies or errors to provide opportunities for data cleaning exercises and to illustrate common data quality issues.*

should we leave the [dataset](https://github.com/vega/vega-datasets/blob/main/data/monarchs.json) as-is and reference the errors in `SOURCES.md`? Or should we correct the errors? Personally, I favor hosting accurate data as the default. In this case, it would be simple for an instructor to reintroduce errors in the dataset if needed for teaching purposes. Then again, correcting the errors may "break" existing examples that rely on the existence of errors.

```diff
[
-{"name":"Elizabeth","start":1565,"end":1603,"index":0},
+{"name":"Elizabeth","start":1558,"end":1603,"index":0},
{"name":"James I","start":1603,"end":1625,"index":1},
{"name":"Charles I","start":1625,"end":1649,"index":2},
{"name":"Cromwell","start":1649,"end":1660,"commonwealth":true,"index":3},
{"name":"Charles II","start":1660,"end":1685,"index":4},
{"name":"James II","start":1685,"end":1689,"index":5},
{"name":"W&M","start":1689,"end":1702,"index":6},
{"name":"Anne","start":1702,"end":1714,"index":7},
{"name":"George I","start":1714,"end":1727,"index":8},
{"name":"George II","start":1727,"end":1760,"index":9},
{"name":"George III","start":1760,"end":1820,"index":10},
-{"name":"George IV","start":1820,"end":1820,"index":11}
+{"name":"George IV","start":1820,"end":1830,"index":11}
]
```

### PR DESCRIPTION
**Pull request by dsmedia**
Resolves #595

Before approving, please review how the dataset corrections will alter the example visualization `Wheat and Wages` appearing in the [Vega](https://vega.github.io/vega/examples/wheat-and-wages/), [Vega-Lite](https://vega.github.io/vega-lite/examples/wheat_wages.html) and [Altair](https://altair-viz.github.io/gallery/wheat_wages.html) example galleries. 

[Original](https://vega.github.io/vega-lite/examples/wheat_wages.html) from vega-lite gallery
[Modified](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAdxoATemgCcABh1L4NMljQAWPUrVJMKVKAYAnQWhBWbBZfGtSZ8gL5KmA5IshAAZmwOCGgA2qBQSIJQDILWcC4A1G4MCJQAnnBIDgAEGcViiiBIEC4FRQD6cLJqIH4AukpNUGxqNLJkaKB5gyCYeTjp6ACODCGYdNY0pJVIAB40NXYgAF59anCraACMfgEgh1tjEy4zcwvzy0prGyPzUADWAMJsDLKYaGIlBEotYXC1TmdUgUHLFQMgHO9XuNJiAmEVKmEaIJnOgAMRIAmVCBBNjvFG4rSU1qdWTdXr9EYXUCYuCCFroOowpRMPouHmyWRwcFKVYAJhGLLZtUKDkazWpIGGW0l7JUXn+EIUoDctlAxESDDgmziiplLiOADY9NShmb0JaxDobab0faAOzWgK210gI5ux3OznmgAcnraZ3hiMuyJcDhS6SUxIcpPJeidiZJZIA6upNKgdJQLUo2DgkFA6EqC2Izl0en0BlsmSAVdL0Zq4UUo6AriiioVKt1BJEXPjTFAhUxKiWyxW0AW3TXaXWGY2JTRWaqg2clcz11L0MokGQja1TlqQJGkdd0MJBQO2EOYXi01PS+WxnPKAuaXT64y1xurYwtuAH7ioR4nu2F6dleKK3gmioAPJhGEEBwP8qD8KKSiDsOeIAKL4cGRzBuKi6-iuoBNi2HJmiByp7qqh7HjUUE6iM+qCIaxqgLISAICi+HCNsSBMOhMDFAAkkSNgOBhRxiGIwY0uylo6AAzEoewHHOXogHxAkuAAUvxRpSTJRTyVamkgE0qkWqKgIgNpFwnOeBkop8MBFIIZnSRmlnHA5Tl2UFphaFpzQ6agop6R5LifMmCDKKyzgBXJYURbZ8qoJaVo4WwCAIHIKWJHmQSGpF+wXOpcWmQl3lOH5-kgMSgW5Ra+XZfZwZOS5Zh1YZ6AmQJEBSS1bUZR1vUqUFwbKc5UUXNW7n1egWYAGQALIWVNlrBlloW5R62GLdVaAWoNKIAIICghk3ySds3HUcphVdF368WtIAAOJwJEx7melj2vc9fqim670XMGV0uH9ANwONu2PRDYNul1-WoFosPoPDDiA5JhPI8c6P6N1xykWTmNHDoOO-f9+OI5JABqxO5ZTYPBupVNLccJzhoEwShMC0SoCaCRJCkaQuAAFDLACE2S5N0RUlYUgj0MUG0bcUSusEtxQAKTFKKACUxQAPzFPwRyoMURzmwAVCbpTFFoFRPJsIBsChaEaueEvJKkmAou7Kxe8M7RKJe0bXiADhwFA-wZsmZIji+5HLg2VGgaqD0KmKucuKF9G7oBHIKnk4oMeX3u++hCqYtiRfoCrxWyKVGswESEuTKAQvHrEqKpB8lSeCIIACyAvnHjlsgpIIbHWLqICODirjLwQ7dFLAEA+HICpBCE4SRKL4uJEH0voHLivWDklBt2rZUSdrut37kLlGyb5tWzbdsO67MOnsXA+zCGRAOF8pYh0yHrB6rsZZZHfpQOy1tSiwNkpgU2BBxTAPQIcKO0EESwRcCHVYyckL1yskCOQmAADKNBtiTCymiNC8E+RsGwIVDENDaFjF8i4BYwgoAKlrPSbO5wW4SNLs2RiICULgNGAcDCZcwLxQhB0EA3RZCYnEc8TY3Y6D8NQPPbESgyAOHUN8R8Lhx7QLMRYtQiE3yznzJQRylhCpIF5KgMIiQ0JnGIOuZQIwkwphIULCApYE5-FPH4IAA) by hard-coding proposed changes

### CODE REVIEW

**Comment by domoritz** [comment]
D3 example: https://mbostock.github.io/protovis/ex/wheat-full.html
Protovis: https://mbostock.github.io/protovis/ex/wheat.html

Original image from Playfair:
image

**Review by domoritz — CHANGES_REQUESTED** [review]
* Having Elizabeth start earlier makes the Vega-Lite chart extend the domain. We should fix the domain before accepting this change. Similar with the end. 

I suspect the data was made specifically for this chart so the start and end doesn't really matter. If we want to change the data, we should update the examples in Vega, Vega-Lite, and Altair before merging.

**Comment by dsmedia** [comment]
It's simple enough to avoid any breaking changes in any dependent visualizations by revising the SOURCES.md file to acknowledge that "the dataset contains two intentional inaccuracies to maintain compatibility with" Wheat and Wages. I didn't include links to the D3 or protovis examples but please feel free to do so if you'd like.

File: data/monarchs.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [
-{"name":"Elizabeth","start":1565,"end":1603,"index":0},
+{"name":"Elizabeth I","start":1565,"end":1603,"index":0},
```
Comment (domoritz) [inline data/monarchs.json:2]: When this dataset was made, only Elisabeth was needed to fully qualify who is meant.

File: data/monarchs.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 {"name":"James I","start":1603,"end":1625,"index":1},
 {"name":"Charles I","start":1625,"end":1649,"index":2},
 {"name":"Cromwell","start":1649,"end":1660,"commonwealth":true,"index":3},
 {"name":"Charles II","start":1660,"end":1685,"index":4},
-{"name":"James II","start":1685,"end":1689,"index":5},
+{"name":"James II","start":1685,"end":1688,"index":5},
 {"name":"W&M","start":1689,"end":1702,"index":6},
```
Comment (domoritz) [inline data/monarchs.json:8]: https://www.hrp.org.uk/kensington-palace/history-and-stories/william-iii-and-mary-ii/#gs.e3lczb says 

William III and Mary II were England’s first and only joint sovereigns, with Mary sharing equal status and power. William and Mary came to the throne after the "Glorious Revolution" of 1688 when Mary’s father, James II, was deposed for trying to enforce Catholic tolerance in England. The King and Queen ruled jointly from **1689** until Mary’s death aged 32 in **1694**.

Is 1702 correct?

**Review by domoritz — CHANGES_REQUESTED** [review]

File: data/monarchs.json (hunk 2 above)
Comment (dsmedia) [reply data/monarchs.json:8]: While William & Mary's joint rule ended in 1694, William's solo reign continued until his death in 1702, when he was thrown from a horse and broke his collar bone.  This is noted in the proposed SOURCES.md entry as follows:

>#### William & Mary's Reign
>The entry "W&M" represents the joint reign of William III and Mary II. While the dataset shows their reign as 1689-1702, the official Web site of the British royal family indicates that Mary II's reign ended in 1694, though William III continued to rule until 1702.

File: SOURCES.md [hunk 3]
```diff
@@ -209,6 +209,46 @@ Selected rail lines simplified from `tfl_lines.json` at https://github.com/oobri
...
+
+#### Interregnum Period
+The `commonwealth` field is used to flag the period from 1649 to 1660, which includes the Commonwealth of England, the Protectorate, and the period leading to the Restoration. While historically more accurate to call this the "interregnum," the field name of `commonwealth` from the original dataset is retained for backwards compatibility.
+
+#### Recent updates
+
+The dataset was revised in Aug. 2024 with the following corrections:
+
+* Elizabeth I's name is updated to include the [regnal number](https://en.wikipedia.org/wiki/Regnal_number) "I"
+* James II's reign now ends in 1688 (previously 1689)
```
Comment (domoritz) [inline SOURCES.md:242]:
```suggestion
The dataset was revised in Aug. 2024. James II's reign now ends in 1688 (previously 1689).
```

**Review by domoritz — APPROVED** [review]
Thanks for the edits. Just one change.

**Pull request merged**